contracts/TsteemPool.sol

```diff
@@ -0,0 +1,289 @@
+pragma solidity ^0.5.0;
+
+import "./SafeMath.sol";
+import "./Ownable.sol";
+
+interface ERC20Token {
+    function transfer(address _to, uint256 _value) external returns (bool);
+
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) external returns (bool);
+
+    function balanceOf(address account) external view returns (uint256);
+}
+
+interface PeanutsPool {
+    function withdrawPeanuts() external;
+
+    function getPendingPeanuts() external view returns (uint256);
+}
+
+contract TsteemPool is Ownable {
+    using SafeMath for uint256;
+    struct Delegator {
+        uint256 tsteemAmount; // deposted tsteem
+        uint256 availablePeanuts; // minted peanuts for user
+        uint256 debtRewards; // rewards debt
+        bool hasDeposited; // set true when first time deposit
+    }
+    mapping(address => Delegator) public delegators;
+    address[] public delegatorsList;
+    uint256 public shareAcc;
+    uint256 public totalDepositedTSTEEM;
+    address public devAddress;
+    uint256 public lastRewardBlock;
+
+    ERC20Token Pnuts;
+    ERC20Token Tsteem;
+    PeanutsPool PnutPool;
+
+    event DepositTSTEEM(address delegator, uint256 amount);
+    event WithdrawTSTEEM(address delegator, uint256 amount);
+    event WithdrawPeanuts(address delegator, uint256 amount);
```

Comment:
Event  name duplicate with PeanutsPool, it would be better we use a different name(e.g. TSteemWithdrawPeanuts)

Reply:
It's the same event name in all peanut contracts.I think it's OK.